allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -177,139 +276,153 @@ def _tokenize(self, sentence_1: str, sentence_2: str = None):
                     continue
                 allowed_skipped_whitespace = min_allowed_skipped_whitespace
 
-                tokens[token_index].idx = token_start_index
+                token_offsets[token_index] = (
+                    token_start_index,
+                    token_start_index + len(token_text),
+                )
                 text_index = token_start_index + len(token_text)
                 token_index += 1
 
-        return tokens
+        tokens = []
+        for token_id, token_type_id, offsets in zip(token_ids, token_type_ids, token_offsets):
+            if offsets is None or offsets[0] >= offsets[1]:
+                token_str = self.tokenizer.convert_ids_to_tokens(
+                    token_id, skip_special_tokens=False
+                )
+                start = None
+            else:
+                start, end = offsets
+                token_str = text[start:end]
 
-    def tokenize_sentence_pair(self, sentence_1: str, sentence_2: str) -> List[Token]:
```

Comment:
I don't see this added back anywhere.  Is the functionality just removed, or did I miss it?

Reply:
Yes, that function is removed. You should now call `tokenize()` and `add_special_tokens()` separately.

Reply:
Ah, I get it now, thanks.

Reply:
At some point, I was tempted to remove the `_add_special_tokens` flag completely, assume it's always true, and then deal with special tokens in a `concatenate()` call that does the right thing. But I decided that is too big a change right now.

Reply:
I agree with that change, and I don't think it's very large (not saying that it should be done in this PR, though).  We'll it removes an argument, so it's a breaking change, ok...

Reply:
Yes, it's not big in terms of lines-of-code, but in downstream impact. I'd have to check and fix all of the readers to be compatible with that change.